tiledb/type/range/range.h

```diff
@@ -132,6 +133,13 @@ class Range {
     return range_.empty() ? nullptr : range_.data();
   }
 
+  /** Returns the pointer to the range as flattend bytes of the requested type.
+   */
+  template <typename T>
+  inline const T* typed_data() const {
```

Comment:
Flagging for discussion: IMO we should add at least an assertion here that the width of the requested type matches the range size, possibly an unconditional check depending on the expected cost.